lib/Controller/ShareController.php

```diff
@@ -36,33 +36,26 @@
 
 class ShareController extends ApiController {
 	private $userId;
-	private $activityService;
-	private $groupManager;
-	private $userManager;
-	private $vaultService;
-	private $shareService;
-	private $credentialService;
-	private $notificationService;
-	private $fileService;
 	private $settings;
 	private $manager;
 
 	private $limit = 50;
 	private $offset = 0;
 
-	public function __construct($AppName,
-	                            IRequest $request,
-	                            $UserId,
-	                            IGroupManager $groupManager,
-	                            IUserManager $userManager,
-	                            ActivityService $activityService,
-	                            VaultService $vaultService,
-	                            ShareService $shareService,
-	                            CredentialService $credentialService,
-	                            NotificationService $notificationService,
-	                            FileService $fileService,
-	                            SettingsService $config,
-	                            IManager $IManager
+	public function __construct(
+		$AppName,
+		IRequest $request,
+		$UserId,
+		private IGroupManager $groupManager,
+		private IUserManager $userManager,
+		private ActivityService $activityService,
+		private VaultService $vaultService,
+		private ShareService $shareService,
+		private CredentialService $credentialService,
+		private NotificationService $notificationService,
+		private FileService $fileService,
+		SettingsService $config,
+		IManager $IManager,
```

Comment:
same here, is there a reason leaving SettingsService and IManager as it is?

Reply:
I wasn't sure whether it breaks the DI or not,  I've addressed it in the next commit